src/Sentry/Internal/BatchProcessorTimer.cs

```diff
@@ -0,0 +1,61 @@
+using System.Timers;
+
+namespace Sentry.Internal;
+
+internal abstract class BatchProcessorTimer : IDisposable
+{
+    protected BatchProcessorTimer()
+    {
+    }
+
+    public abstract bool Enabled { get; set; }
+
+    public abstract event EventHandler<ElapsedEventArgs> Elapsed;
+
+    public void Dispose()
+    {
+        Dispose(true);
+        GC.SuppressFinalize(this);
+    }
+
+    protected virtual void Dispose(bool disposing)
+    {
+    }
+}
+
+internal sealed class TimersBatchProcessorTimer : BatchProcessorTimer
+{
+    private readonly System.Timers.Timer _timer;
+
+    public TimersBatchProcessorTimer(TimeSpan interval)
+    {
+        _timer = new System.Timers.Timer(interval.TotalMilliseconds)
+        {
+            AutoReset = false,
+            Enabled = false,
+        };
+        _timer.Elapsed += OnElapsed;
+    }
+
+    public override bool Enabled
```

Comment:
we're leaking the .NET API here and it's a pretty bad API. Using a property to set something with a side effect as a timer is not ideal.

Lets call this `Start()` and `Stop()` to be very clear what it does

Reply:
Abstraction deleted: see https://github.com/getsentry/sentry-dotnet/pull/4310#discussion_r2177748194